frontend/src/metabase/query_builder/components/view/sidebars/SummarizeSidebar/SummarizeContent/SummarizeAggregationItemList.tsx

```diff
@@ -1,43 +1,62 @@
+import { useMemo } from "react";
+
+import type { UpdateQueryHookProps } from "metabase/query_builder/hooks/types";
+import { getAggregationItems } from "metabase/query_builder/utils/get-aggregation-items";
 import { Group, type GroupProps } from "metabase/ui";
-import type * as Lib from "metabase-lib";
+import * as Lib from "metabase-lib";
 
 import { AddAggregationButton } from "../AddAggregationButton";
 import { AggregationItem } from "../AggregationItem";
 
-type SummarizeAggregationItemListProps = {
-  query: Lib.Query;
-  stageIndex: number;
-  aggregations: Lib.AggregationClause[];
-  onQueryChange: (query: Lib.Query) => void;
-} & GroupProps;
+type SummarizeAggregationItemListProps = UpdateQueryHookProps & GroupProps;
 
 export const SummarizeAggregationItemList = ({
   query,
-  stageIndex,
-  aggregations,
   onQueryChange,
+  stageIndex,
   ...containerProps
-}: SummarizeAggregationItemListProps) => (
-  <Group
-    data-testid="summarize-aggregation-item-list"
-    spacing="sm"
-    align="flex-start"
-    {...containerProps}
-  >
-    {aggregations.map((aggregation, aggregationIndex) => (
-      <AggregationItem
-        key={aggregationIndex}
+}: SummarizeAggregationItemListProps) => {
+  const aggregationItems = useMemo(
+    () =>
+      getAggregationItems({
+        query,
+        stageIndex,
+      }),
+    [query, stageIndex],
+  );
```

Comment:
nit: formatting, less scrolling = better

```suggestion
  const aggregationItems = useMemo(
    () => getAggregationItems({ query, stageIndex }),
    [query, stageIndex],
  );
```